### ISSUE TITLE
Using KestrelCommunicationListener in simple Service Farbic Application results in 410 (Gone)

### ISSUE THREAD

**Issue by jsacapdev** [Azure/service-fabric-aspnetcore#13]
Minimal repro steps

1. Open Visual Studio 2017
2. Create new project -> Cloud -> Service Fabric Application -> ASP.NET Core -> Empty 
3. In Web1.cs, replace CreateServiceInstanceListeners with:

        protected override IEnumerable<ServiceInstanceListener> CreateServiceInstanceListeners()
        {
            return new ServiceInstanceListener[]
            {
                new ServiceInstanceListener(serviceContext =>
                    new KestrelCommunicationListener(serviceContext, "ServiceEndpoint", (url, listener) =>
                    {
                        return new WebHostBuilder()
                            .UseKestrel()
                            .ConfigureServices(
                                services => services
                                    .AddSingleton<StatelessServiceContext>(serviceContext))
                            .UseContentRoot(Directory.GetCurrentDirectory())
                            .UseServiceFabricIntegration(listener, ServiceFabricIntegrationOptions.UseUniqueServiceUrl)
                            .UseStartup<Startup>()
                            .UseUrls(url)
                            .Build();
                    }))
            };
        }

Expected result

Can hit breakpoint in Startup.cs:

app.Run(async (context) =>
{
	await context.Response.WriteAsync("!!!");
});

Actual result

PS C:\> invoke-webrequest http://localhost:8880/
invoke-webrequest : The remote server returned an error: (410) Gone.
At line:1 char:1
+ invoke-webrequest http://localhost:8880/
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

Further technical details

Windows 10 Anniversary Edition

Microsoft Visual Studio Enterprise 2017 
Version 15.1 (26403.7) Release
VisualStudio.15.Release/15.1.0+26403.7
Microsoft .NET Framework
Version 4.6.01586

Service Fabric: 5.5.216.0
Service Fabric SDK:	2.5.2160.0
Microsoft.ServiceFabric.AspNetCore.Kestrel: 2.5.2160.0

Endpoint Configuration:

<Endpoints>
  <Endpoint Protocol="http" Name="ServiceEndpoint" Port="8880" />
</Endpoints>

**Comment by amanbha** [comment]
@jsacapdev 
That's expected behavior with ServiceFabricIntegrationOptions.UseUniqueServiceUrl, so the actual url reported by the listener to ServiceFabric runtime will not be http://localhost:8880/, It will have some guilds appended to it to uniquely identify the service, to get this url you will have to resolve the service by using Service Fabric client apis or for your dev box you can look up the url as reported to Service fabric runtime in Service Fabric Explorer.
If you want to use static ports, just use ServiceFabricIntegrationOptions.None.
For more info:, please see https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-communication-aspnetcore#service-fabric-integration-middleware

**Comment by jsacapdev** [comment]
Thanks.

**Comment by mtorkamani** [comment]
Hey @amanbha,
I'm having the same issue. I deployed a stateful  ASPNetCore api service and want to connect to that via HttpCilent from another service in the cluster. How can I get its Url? 
I find my deployed service in SF explorer and see my api service has been assigned a port dynamically and last segment of it's endpoint is a Guid which is added by Integration middleware (Kestrel). I followed this sample at [https://github.com/Azure-Samples/service-fabric-dotnet-getting-started/tree/classic/Services/WordCount](url). But the URL I get doesn't have the last Guid segment. Any idea?

Cheers,
Mehdi
